CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+cmake_minimum_required(VERSION 3.19)
+
+project(finufft VERSION 2.1.0 LANGUAGES C CXX)
+
+set(GNU_LIKE_FRONTENDS AppleClang Clang GNU)
+if(CMAKE_CXX_COMPILER_ID IN_LIST GNU_LIKE_FRONTENDS)
```

Comment:
Instead of checking the compiler I think the best way is to use
```cmake
  foreach (FLAG "-O3" "-DNDEBUG" "-funroll-loops")
    unset (SUPPORT_FLAG CACHE)
    unset (SUPPORT_FLAG )
    check_c_compiler_flag (${FLAG} SUPPORT_FLAG) #check_cxx_compiler_flag for c++
    if (SUPPORT_FLAG)
      list(APPEND CMAKE_CXX_FLAGS_RELEASE ${FLAG})
    endif ()
  endforeach ()
```